influxql/parser.go

```diff
@@ -493,6 +503,63 @@ func (p *Parser) parseSelectStatement() (*SelectStatement, error) {
 	return stmt, nil
 }
 
+// targetRequirement specifies whether or not a target clause is required.
```

Comment:
I hear ya -- I too don't like seeing random `true`s tacked onto function signatures. I tend to instead do something like:
```
parseStringTargetRequired()
```
and
```
parseStringTargetNotRequired()
```